python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1392,8 +1392,16 @@ def external_asset_graph_from_defs(
                     job_def.name
                     for job_def in job_defs
                     if source_asset.key in job_def.asset_layer.source_assets_by_key
-                    and source_asset.partitions_def is None
-                    or source_asset.partitions_def == job_def.partitions_def
+                    and (
+                        # explicit source-asset observation job
+                        not job_def.asset_layer.has_assets_defs
+                        # "base asset job" will have both source and materializable assets
+                        or is_base_asset_job_name(job_def.name)
```

Comment:
if I'm understanding correctly, the code here is just to ensure that source assets will reference the "base asset job" in the list of jobs they are a part of. This doesn't seem necessary, as we don't display this job to the user (it's just a trick to let us do ad-hoc materializations). In the interest of simplicity, would prefer to just remove this

Reply:
This _is_ necessary for the "Observe" button to work correctly when you select an observable in the Asset Graph Explorer. Observations launched in this way are executed as a selection on one of these base asset jobs.

Reply:
ah makes sense